lib/mayaUsd/fileio/jobs/writeJob.cpp

```diff
@@ -1375,6 +1382,53 @@ bool UsdMaya_WriteJob::_CheckNameClashes(const SdfPath& path, const MDagPath& da
     return true;
 }
 
+void UsdMaya_WriteJob::_AddDefaultPrimAccessibility()
+{
+    auto defaultPrim = mJobCtx.mStage->GetDefaultPrim();
+    if (!defaultPrim) {
+        return;
+    }
+
+    auto accessibilityLabel = mJobCtx.mArgs.accessibilityLabel;
+    auto accessibilityDescription = mJobCtx.mArgs.accessibilityDescription;
+    if (accessibilityLabel.empty() && accessibilityDescription.empty()) {
+        return;
+    }
+
+    /* The USD AccessibilityAPI is only available from OpenUSD 25.5 onwards.
+     * We support writing the data with ad-hoc attributes on pre-25.5 versions,
+     * and use the actual API for 25.5 and beyond.
+     */
+#if PXR_VERSION >= 2505
+    auto api = UsdUIAccessibilityAPI::ApplyDefaultAPI(defaultPrim);
+    if (!accessibilityLabel.empty()) {
+        defaultAPI.CreateLabelAttr(VtValue(accessibilityLabel));
+    }
+    if (!accessibilityDescription.empty()) {
+        defaultAPI.CreateLabelAttr(VtValue(accessibilityDescription));
+    }
+#else
+    defaultPrim.AddAppliedSchema(TfToken("AccessibilityAPI:default"));
+    if (!accessibilityLabel.empty()) {
+        auto labelAttr = defaultPrim.CreateAttribute(
+            TfToken("accessibility:default:label"),
+            SdfValueTypeNames->String,
+            false,
+            SdfVariabilityVarying);
+        labelAttr.Set(accessibilityLabel);
```

Comment:
Does that work? I somehow thought USD refused to create ttributes that are not explicitly declared in a schema, but maybe I mis-remember. It might be only metadata that need such declaration?

I guess if the tests pass, it means it works?

Reply:
You might be thinking of metadata. USD requires that metadata keys be preregistered (honestly not sure why, because the underlying data doesn't need to care)

But for properties (attributes and relationships) it allows arbitrary creation without registered schemas. It's useful because you can author things that haven't been solidified into schemas yet.